src/dashboard/domain/traffic/traffic-data.php

```diff
@@ -0,0 +1,77 @@
+<?php
+// phpcs:disable Yoast.NamingConventions.NamespaceName.TooLong -- Needed in the folder structure.
+namespace Yoast\WP\SEO\Dashboard\Domain\Traffic;
+
+use Yoast\WP\SEO\Dashboard\Domain\Data_Provider\Data_Interface;
+
+/**
+ * Domain object that represents a single Traffic record.
+ */
+class Traffic_Data implements Data_Interface {
+
+	/**
+	 * The sessions.
+	 *
+	 * @var int $sessions
+	 */
+	private $sessions;
+
+	/**
+	 * The total users.
+	 *
+	 * @var int $total_users
+	 */
+	private $total_users;
+
+	/**
+	 * The constructor.
+	 *
+	 * @param int $sessions    The sessions.
+	 * @param int $total_users The total users.
+	 */
+	public function __construct( ?int $sessions = null, ?int $total_users = null ) {
```

Comment:
Does it make sense to allow these to be able to be null? And if that is the case the comments needs to show that? Is the object still valuable with only a total number of users?

Reply:
These are able to be null because we might create a Traffic Data record with just sessions, just users, or both. So yeah, an object is still valuable with only a total number of users. An object is only going to be useless if both sessions and users are null, do you think I should safeguard against that (maybe with an exception in the constructor)?

Reply:
> An object is only going to be useless if both sessions and users are null, do you think I should safeguard against that (maybe with an exception in the constructor)?

On a second thought, we initially create an object with no metrics and later populate the right one, so let's just remove the constructor and leave the setters of each metric in.